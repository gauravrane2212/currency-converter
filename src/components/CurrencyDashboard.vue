<template>
  <div class="dashboard">
    <h1>Currency Converter</h1>
    <div class="card">
      <CurrencyHeader @onInputChanged="valueChanged" />
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from "vue-property-decorator";
import CurrencyHeader from "@/components/CurrencyHeader.vue";

@Component({
  components: {
    CurrencyHeader
  }
})
export default class CurrencyDashboard extends Vue {
  private userEnteredAmount: number = 10;

  valueChanged(newValue: number): void {
    this.userEnteredAmount = newValue;
  }
}
</script>

<style scoped lang="scss">
h1 {
  margin: 40px 0 20px;
}

.card {
  margin: 0 auto;
  width: 540px;
  border: 1px ridge rgb(218, 215, 215);
}
</style>
