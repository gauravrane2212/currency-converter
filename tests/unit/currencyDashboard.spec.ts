import { shallowMount } from "@vue/test-utils";
import CurrencyDashboard from "@/components/CurrencyDashboard.vue";

describe("CurrencyDashboard.vue", () => {
  it("should match current snapshot", () => {
    const wrapper = shallowMount(CurrencyDashboard);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
