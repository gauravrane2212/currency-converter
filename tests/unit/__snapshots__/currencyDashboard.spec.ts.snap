// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyDashboard.vue should match current snapshot 1`] = `
<div class="dashboard">
  <h1>Currency Converter</h1>
  <div class="card">
    <currencyheader-stub></currencyheader-stub>
  </div>
</div>
`;
